core-api/src/main/java/com/optimizely/ab/notification/NotificationCenter.java

```diff
@@ -78,6 +82,43 @@ public NotificationCenter() {
     // we used a list so that notification order can mean something.
     private Map<NotificationType, ArrayList<NotificationHolder>> notificationsListeners =new HashMap<NotificationType, ArrayList<NotificationHolder>>();
 
+    /**
+     * Convenience method to support lambdas as callbacks in later version of Java (8+).
+     * @param activateNotificationListenerInterface
+     * @return greater than zero if added.
```

Comment:
I wonder if we will run into concurrency/threading issues with incrementing this static variable. What if we end up with the same ID for different listeners?

Reply:
not static.  you are looking at the holder class def and not the notification id.